app/page.tsx

```diff
@@ -9,90 +11,39 @@ import { useState } from 'react';
-	// 			).value.toLowerCase();
-
-	// 			try {
-	// 				// Step 5: Send a fetch POST Call to the backend api point /api/pokemons which will then trigger a fetch to the pokemon api
-	// 				// go to the route.tsx to continue the lesson
-	// 				const res = await fetch('/api/pokemons', {
-	// 					method: 'POST',
-	// 					body: JSON.stringify({ name: pokemonName }),
-
-	// 					headers: {
-	// 						'Content-Type': 'application/json',
-	// 					},
-	// 				});
-
-	// 				// Step 11: we are now back from the backend, api and if the res is NOT okay, then we shall throw an error and the http status
-	// 				if (!res.ok) {
-	// 					throw new Error(`HTTP error! Status: ${res.status}`);
-	// 				}
-
-	// 				// Step 12: We are also in an async function, and we are awaiting the response from the backend to be OK or status 200
-	// 				const responseData = await res.json();
-
-	// 				// Step 13: a checker to make sure we have a name within the response Object
-	// 				if (responseData.name) {
-	// 					// Step 14: if the object has a name, then we shall redirect the user to the pokemon page
-	// 					const redirectUrl = `/pokemons/${responseData.name}`;
-	// 					window.location.href = redirectUrl;
-	// 				} else {
-	// 					console.error('Invalid response data');
-	// 				}
-	// 			} catch {
-	// 				new Error('Pokemon Not Valid');
-	// 			}
-	// 		});
-	// 	}
-	// }
-
 	const [pokemon, setPokemon] = useState<null | { name: string }>(null);
 	const [errorMessage, setErrorMessage] = useState('');
 
 	const onSubmit = async (event) => {
 		event.preventDefault();
 		const formData = new FormData(event.currentTarget);
 		const pokemonName = formData.get('pokemonName');
-		// formData.append('pokemonName', pokemonName);
 
 		try {
 			console.log('pokemonNameFrontend', pokemonName);
 			const res = await fetch('/api/pokemons', {
 				method: 'POST',
 				body: formData,
-				// headers: {
-				// 	'Content-Type': 'application/json',
-				// },
 			});
 
 			// Step 11: we are now back from the backend, api and if the res is NOT okay, then we shall throw an error and the http status
 			if (!res.ok) {
+				console.log('Somethings WRONG');
 				throw new Error(`HTTP error! Status: ${res.status}`);
 			}
 
 			// Step 12: We are also in an async function, and we are awaiting the response from the backend to be OK or status 200
 			const responseData = await res.json();
 
-			console.log('responseDataFRONT', responseData);
 			// Step 13: a checker to make sure we have a name within the response Object
 			if (responseData.pokemonData) {
-				// Step 14: if the object has a name, then we shall redirect the user to the pokemon page
-				// const redirectUrl = `/pokemons/${responseData.pokemonData.name}`;
-				// window.location.href = redirectUrl;
-
 				setPokemon(responseData.pokemonData);
+				// redirect(`http://localhost:3000/pokemons/${responseData.pokemonData.name}`);
+
+				const resetForm = document.getElementById('form');
```

Comment:
The build doesn't like this resetForm.reset()

Reply:
Type error: 'resetForm' is possibly 'null'.

tried chaining
like resetForm?.reset()